src/polygon.jl

```diff
@@ -68,19 +68,62 @@ end
 
 
 """
-    get(code)
-Returns the MULTIPOLYGON data for the requested country.\n
-Input: ISO 3166 Alpha 3 Country Code
+    get(country, levels...)
+Returns the MULTIPOLYGON data for the requested region.\n
+Input: ISO3 Country Code, and further subdivisions\n
+
+Examples:\n
+1. `get("IND")` - Returns polygon of India
+2. `get("IND", "Uttar Pradesh")` - Returns polygon of the state Uttar Pradesh
+3. `get("IND", "Uttar Pradesh", "Lucknow")` - Returns polygon of district Lucknow\n
 """
-function get(code)
+function get(country, levels...) 
     # only uppercase country codes are accepted
-    isvalidcode(code) || throw(ArgumentError("please provide standard ISO 3166 Alpha 3 country codes"))
+    isvalidcode(country) || throw(ArgumentError("please provide standard ISO 3 country codes"))
 
-    dataid = "GADM_$code"
+    dataid = "GADM_$country"
 
     resource_data_path = download(dataid)
 
     dataset = extractdataset(resource_data_path)
 
-    extractgeometry(dataset)
+    # the number of varargs in levels is the required level for GADM dataset
+    required_level = length(levels)
+
+    # get layer of the required level
+    level = getlevel(dataset, required_level)
+
+    # get country level
+    if required_level == 0
+        return ArchGDAL.getfeature(level, 1) do feature
+            ArchGDAL.getgeom(feature)
+        end
+    end
+
+    nfeatures = ArchGDAL.nfeature(level)
```

Comment:
I am bit confused with this logic. So you find out what is the level based on the number of arguments, I get it. Then why the following code has a different behavior for countries compared to other levels? Shouldn't the code look the same after you find out the current level of the query?

Reply:
I had initially written the same code for all levels. But then I realised that names need not be compared while getting the polygon for a country. While in the logic implemented below, features were being filtered by the names. So I brought this logic out of the loop.